eth/beacon/state_machines/forks/serenity/active_states.py

```diff
@@ -1,14 +1,25 @@
+
+from typing import (
+    Sequence,
+    TYPE_CHECKING,
+)
+
 from eth.beacon.types.active_states import ActiveState
 from eth.beacon.helpers import (
     get_new_recent_block_hashes,
 )
 
+if TYPE_CHECKING:
+    from eth.beacon.types.blocks import BaseBeaconBlock  # noqa: F401
+
 
 class SerenityActiveState(ActiveState):
     @classmethod
-    def from_old_active_and_blocks(cls, old_active_state, blocks, recent_block_hashes_length):
-        recent_block_hashes = old_active_state.recent_block_hashes
-        pending_attestations = old_active_state.pending_attestations
+    def from_backup_active_state_and_blocks(cls,
+                                            backup_active_state: ActiveState,
+                                            blocks: Sequence['BaseBeaconBlock']) -> ActiveState:
```

Comment:
:+1: to using `Sequence`.  I think we (and largely me) have been defining these as `Tuple` or `List` too often which isn't strictly necessary and doesn't provide much extra safety.

Reply:
When judging between `Iterable` and `Sequence` my preference is to always use the most flexible type that fulfills the goal. 

For example, if you do not need to access `len()` (or e.g index access), `Iterable` will do it and is more flexible. For instance, it allows you to also pass in an iterator (`yield "foo"`). 

But if you need to get the length or need to access elements at a specific index, then one has to tighten the type to something like `Sequence` because `Iterable` simply can't give you that (e.g. it is perfectly valid for an `Iterable` to be infinite, so not having an answer to `len()` at all).

Reply:
@cburgdorf thank you for elaborating! I will keep the current `Iterable`s.